### PR TITLE
add support to export the full transcript (VCA + message_m)

### LINKED ISSUE #29: add support to export the full transcript (VCA + message_m) for GET_MEASUREMENT
This is to allow external entity to validate the nonce associated with the measurement.

### PR DESCRIPTION
**Pull request by longlongyang**
fix #29

### CODE REVIEW

File: spdmlib/src/requester/get_measurements_req.rs [hunk 1]
```diff
@@ -15,28 +15,40 @@ use crate::protocol::*;
...
 impl RequesterContext {
+    #[allow(clippy::too_many_arguments)]
     async fn send_receive_spdm_measurement_record(
         &mut self,
         session_id: Option<u32>,
         measurement_attributes: SpdmMeasurementAttributes,
         measurement_operation: SpdmMeasurementOperation,
         spdm_measurement_record_structure: &mut SpdmMeasurementRecordStructure,
+        transcript_vca: &mut Option<ManagedBufferA>,
+        transcript_meas: &mut Option<ManagedBufferM>,
```
Comment (jyao1) [inline spdmlib/src/requester/get_measurements_req.rs:26]: If the caller needs to invoke send_receive_spdm_measurement_record() multiple times, then transcript_vca will be copied multiple times.

That is not efficient.

Can we do that in init_connection()?

Comment (longlongyang) [reply spdmlib/src/requester/get_measurements_req.rs:26]: > If the caller needs to invoke send_receive_spdm_measurement_record() multiple times, then transcript_vca will be copied multiple times.
> 
> That is not efficient.
> 
> Can we do that in init_connection()?

Sure, updated.

File: spdmlib/src/requester/get_measurements_req.rs [hunk 2]
```diff
@@ -173,6 +184,14 @@ impl RequesterContext {
...
+                            ret_message_m
+                                .append_message(send_buffer)
+                                .ok_or(SPDM_STATUS_BUFFER_FULL)?;
+                            ret_message_m
+                                .append_message(&receive_buffer[..temp_used])
+                                .ok_or(SPDM_STATUS_BUFFER_FULL)?;
+                        }
 
                         // verify signature
                         if measurement_attributes
```
Comment (jyao1) [inline spdmlib/src/requester/get_measurements_req.rs:218]: Need append signature

**Comment by jyao1** [comment]
Please also resolve conflict

File: spdmlib/src/requester/get_measurements_req.rs (hunk 2 above)
Comment (longlongyang) [reply spdmlib/src/requester/get_measurements_req.rs:218]: > Need append signature

updated.

**Comment by longlongyang** [comment]
> Please also resolve conflict

resolved.

**Comment by jyao1** [comment]
Need test to ensure the digital signature can be verified.

**Comment by longlongyang** [comment]
> Need test to ensure the digital signature can be verified.

new test function added.

**Comment by jyao1** [comment]
Looks good.

We probably need add one more test to cover one-by-one mode.

**Comment by longlongyang** [comment]
> Looks good.
> 
> We probably need add one more test to cover one-by-one mode.

one-by-one mode added.

**Review by jyao1 — APPROVED** [review]

**Pull request merged**